### DISCUSSION TITLE
Ruby's version is not correct

### DISCUSSION THREAD

**Discussion by datawolfdev** [rbenv/rbenv#1548] [Q&A]
I need the version 2.7.3 of ruby to access to a project but when I run "rails s" i have an error "Your Ruby version is 3.0.2, but your Gemfile specified 2.7.3". When I do "ruby --version" or "rbenv version" it told me taht my version is the 2.7.3

Thank you for your help

**Comment by mislav** [comment]
Hi, not sure what exactly caused this, but it could be that running `rails` on your machine executes something other a "rails" shim that is managed by rbenv, which is the mechanism with which rbenv does version-switching.

You can perform these checks:

1. `rbenv version` in your project's directory - you've said that this prints "2.7.3", which is good.

2. `which ruby` - this should print something like `~/.rbenv/shims/ruby`, where "~" represents your home directory. This assumes that you haven't edited the rbenv default by setting RBENV_ROOT explicitly.

3. `which rails` - this should print`~/.rbenv/shims/rails`. If it doesn't, this could be where the problem starts from. To fix this, you could tell rbenv to generate its shims by running `rbenv rehash`, or you could try running `bundle exec rails` to make sure that the `rails` executable is explicitly run in the context of the current Ruby and in the context of the current project's Gemfile.
